### ISSUE TITLE
Interface field Connection.edges

### ISSUE THREAD

**Issue by eliyadzz** [wp-graphql/wp-graphql#2699]
### Description

After updating to     `1.13.7` the following errors appear in GraphiQL
```
"errors": [
    "Interface field Connection.edges expects type [Edge!]! but CurrencyTaxonomyGqlToCasino_reviewConnection.edges is type [CurrencyTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CasinosTaxonomyGqlToAffiliate_programConnection.edges is type [CasinosTaxonomyGqlToAffiliate_programConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but AffiliateSoftwareTaxonomyGqlToAffiliate_programConnection.edges is type [AffiliateSoftwareTaxonomyGqlToAffiliate_programConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CategoryToCasino_gameConnection.edges is type [CategoryToCasino_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but WriterTaxonomyGqlToBonus_typeConnection.edges is type [WriterTaxonomyGqlToBonus_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but BonusTaxonomyGqlToBonus_typeConnection.edges is type [BonusTaxonomyGqlToBonus_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but WriterTaxonomyGqlToCasino_gameConnection.edges is type [WriterTaxonomyGqlToCasino_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but WriterTaxonomyGqlToCasino_reviewConnection.edges is type [WriterTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but PaymentOptionTaxonomyGqlToCasino_reviewConnection.edges is type [PaymentOptionTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but WriterTaxonomyGqlToSlot_gameConnection.edges is type [WriterTaxonomyGqlToSlot_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but SlotTypeTaxonomyGqlToSlot_gameConnection.edges is type [SlotTypeTaxonomyGqlToSlot_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but SlotTypeTaxonomyGqlToSlot_typeConnection.edges is type [SlotTypeTaxonomyGqlToSlot_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but SlotsGameTaxonomyGqlToSlot_gameConnection.edges is type [SlotsGameTaxonomyGqlToSlot_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but SoftwareTaxonomyGqlToCasino_reviewConnection.edges is type [SoftwareTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but SoftwareTaxonomyGqlToSlot_gameConnection.edges is type [SoftwareTaxonomyGqlToSlot_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CategoryToCasino_reviewConnection.edges is type [CategoryToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but TagToCasino_promotionConnection.edges is type [TagToCasino_promotionConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but LicenseTaxonomyGqlToCasino_reviewConnection.edges is type [LicenseTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CountryTaxonomyGqlToCasino_reviewConnection.edges is type [CountryTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CasinosTaxonomyGqlToCasino_reviewConnection.edges is type [CasinosTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CasinoGameTaxonomyGqlToCasino_gameConnection.edges is type [CasinoGameTaxonomyGqlToCasino_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but CasinoGameTaxonomyGqlToCasino_reviewConnection.edges is type [CasinoGameTaxonomyGqlToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToAffiliate_programConnection.edges is type [RootQueryToAffiliate_programConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToBonus_typeConnection.edges is type [RootQueryToBonus_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToCasino_gameConnection.edges is type [RootQueryToCasino_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToCasino_promotionConnection.edges is type [RootQueryToCasino_promotionConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToCasino_reviewConnection.edges is type [RootQueryToCasino_reviewConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToOperating_systemConnection.edges is type [RootQueryToOperating_systemConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSlot_gameConnection.edges is type [RootQueryToSlot_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSlot_typeConnection.edges is type [RootQueryToSlot_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSport_bonus_typeConnection.edges is type [RootQueryToSport_bonus_typeConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSport_bonusConnection.edges is type [RootQueryToSport_bonusConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSport_bookConnection.edges is type [RootQueryToSport_bookConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSport_gameConnection.edges is type [RootQueryToSport_gameConnectionEdge!]!.",
    "Interface field Connection.edges expects type [Edge!]! but RootQueryToSport_tipConnection.edges is type [RootQueryToSport_tipConnectionEdge!]!."
  ]
}
```

### Steps to reproduce

It happens only with existing posts of mine

if you go to `WPConnectionType.php`
and set the next line to `false`
`$this->include_default_interfaces = isset( $config['includeDefaultInterfaces'] ) ? (bool) $config['includeDefaultInterfaces'] : true;`

the errors go away

### WPGraphQL Version

1.13.7

### WordPress Version

6.1.1

### PHP Version

8.0.22

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by justlevine** [comment]
@eliyadzz how are you registering all those custom types / connections?

WPGraphQL v1.13.x fixes connections to ensure they correctly implement the GraphQL Relay spec, and the errors your getting are about those types not following the spec correctly.

**Comment by eliyadzz** [comment]
You understand that I didn't change anything and it worked before.
What is your suggestion?

**Comment by yaronovsky** [comment]
I am not setting connections, only registering posts and taxonomies according to the docs, using single, plural and show in graphql. Perhaps there is taxonomy connection cache in the database?

**Comment by justlevine** [comment]
> You understand that I didn't change anything and it worked before.
> What is your suggestion?

@eliyadzz Yes I _do_ understand that 😇. As I noted v1.13.x handles additional validation that the generated types implement the Relay spec. While this normally shouldn't cause any problems, if your code breaks relay spec, then you'll get the above GraphQL errors.

It seems you already discovered that you can set the new `includeDefaultInterfaces` array key in the CPT/tax args to disable the new enforcement ( [pr](https://github.com/wp-graphql/wp-graphql/pull/2655), but unless you're doing something overtly custom or relying on an outdated 3rd party plugin that shouldn't be necessary, and its probably not ideal to disable since u will lose out on the connection args defined by the applied interfaces.

As such, my suggestion is for you to share the code you're using to register the types / connections - or even better: a specific replicable test case - so we can review the specific code and tell you what you need to be doing differently.

> I am not setting connections, only registering posts and taxonomies according to the docs, using single, plural and show in graphql. Perhaps there is taxonomy connection cache in the database?

@yaronovsky unclear to me if you're chiming in with your own issue, or answering for @eliyadzz . Either way, can you share an example config that you're using for both `register_post_type()` and `register_taxonomy()` that represents one of those problematic connections (e.g. currency + casino_review)?

**Comment by eliyadzz** [comment]
Sorry for the mess, I have 2 profiles

$args = [
		'public'              => true,
		'show_in_rest'        => true,
		'hierarchical'        => true,
		'show_ui'             => true,
		'show_in_graphql'     => true,
		'graphql_single_name' => 'casino_review',
		'graphql_plural_name' => 'casino_reviews',
		'has_archive'         => true,
		'query_var'           => true,
		'rewrite'             => [ 'slug' => 'casinos' ],
		'supports'            => [ 'editor', 'thumbnail', 'title' ],
		'label'               => _( 'Casino Reviews' ),
		'map_meta_cap'        => true
	];
	register_post_type( 'casino_review', $args );


$args = [
	'public'              => true,
	'hierarchical'        => true,
	'show_ui'             => true,
	'graphql_single_name' => 'currencyTaxonomyGql',
	'graphql_plural_name' => 'currenciesTaxonomyGql',
	'show_in_graphql'     => true,
	'label'               => _( CURRENCIES_TAXONOMY_LABEL ),
];
register_taxonomy( 'currencies', [ 'casino_review', 'sportbook', 'bonus', 'country', 'tournament' ], $args );

**Comment by eliyadzz** [comment]
This fixes my issue, all types left the same as before,
I don't do any connection manipulations or custom stuff.
Please advise how to solve it without overriding defaults.

```
add_filter( 'graphql_wp_connection_type_config', function ( $config ) {
	$config['includeDefaultInterfaces'] = false;
	return $config;
}, 10, 1 );
```

**Comment by react-ralle** [comment]
Hi,

i have the same issue with multiple taxonomy. 

```
"Interface field Connection.edges expects type [Edge!]! but AssignmentTeamToTeam_memberConnection.edges is type [AssignmentTeamToTeam_memberConnectionEdge!]!."

$labels = array(
	'name' => _x( 'Zuordnungen', 'taxonomy general name' ),
	'singular_name' => _x( 'Zuordnung', 'taxonomy singular name' ),
	'search_items' =>  __( 'Suche Zuordnung' ),
	'all_items' => __( 'Alle Zuordnungen' ),
	'parent_item' => __( 'Über Zuordnung' ),
	'parent_item_colon' => __( 'Über Zuordnung:' ),
	'edit_item' => __( 'Zuordnung bearbeiten' ), 
	'update_item' => __( 'Update Zuordnung' ),
	'add_new_item' => __( 'Zuordnung erstellen' ),
	'new_item_name' => __( 'Neue Zuordnung' ),
	'menu_name' => __( 'Zuordnungen' ),
);    
  
register_taxonomy('assignment-post',"post", array(
	'hierarchical' => true,
	'labels' => $labels,
	'show_ui' => true,
	'show_in_rest' => true,
	'show_in_graphql' => true,
	'graphql_single_name' => 'assignmentPost',
	'graphql_plural_name' => 'assignmentPosts',
	'show_admin_column' => true,
	'query_var' => true,
	'rewrite' => array( 'slug' => 'assignment-post' ),
));
```

`'includeDefaultInterfaces' => false`  is not working for me and also the following not:

```
add_filter( 'graphql_wp_connection_type_config', function ( $config ) {
	$config['includeDefaultInterfaces'] = false;
	return $config;
}, 10, 1 );
```

Is there an update on this?

**Comment by jasonbahl** [comment]
I'm looking to see if I can reproduce this now 👀

**Comment by jasonbahl** [comment]
Ok, I was able to reproduce this by registering a post type and taxonomy as follows:

```php
add_action( 'init', function() {

	$args = [
		'public'              => true,
		'show_in_rest'        => true,
		'hierarchical'        => true,
		'show_ui'             => true,
		'show_in_graphql'     => true,
		'graphql_single_name' => 'casino_review',
		'graphql_plural_name' => 'casino_reviews',
		'has_archive'         => true,
		'query_var'           => true,
		'rewrite'             => [ 'slug' => 'casinos' ],
		'supports'            => [ 'editor', 'thumbnail', 'title' ],
		'label'               => _( 'Casino Reviews' ),
		'map_meta_cap'        => true
	];
	register_post_type( 'casino_review', $args );


	$args = [
		'public'              => true,
		'hierarchical'        => true,
		'show_ui'             => true,
		'graphql_single_name' => 'currencyTaxonomyGql',
		'graphql_plural_name' => 'currenciesTaxonomyGql',
		'show_in_graphql'     => true,
		'label'               => 'Currency',
	];
	register_taxonomy( 'currencies', [ 'casino_review', 'sportbook', 'bonus', 'country', 'tournament' ], $args );

});
```

This is good news!

I'm not sure yet what the cause is, but I can at least get in the same state. 

Hope to have a fix for this shortly. 🙏🏻

**Comment by jasonbahl** [comment]
Ok, so it looks like this might be related to underscores in the `graphql_single_name` and `graphql_plural_name`. 

When I remove the underscores, this issue goes away. 

🤔

**Comment by jasonbahl** [comment]
I believe I've found the culprit in the WPConnectionType class. 

Running some tests to validate. Hope to have a PR shortly.

**Comment by jasonbahl** [comment]
I've released v1.13.9 with this fix: https://github.com/wp-graphql/wp-graphql/releases/tag/v1.13.9
